File: chronos.py
#!/usr/bin/env python

from time import sleep
from chronos import SimpleChronosSocket

def func(str):
  print "Got data: " +  str

if __name__ == '__main__':
  socket = SimpleChronosSocket("/chronos", func)
  sleep(1)
  socket.publish_string("/1/2", 0, "Hello from socket.", 100)
  sleep(1)
  socket2 = SimpleChronosSocket("/chronos", func)

  sleep(1)
  print "done"

File: chronos.cpp
#include <boost/python.hpp>
#include "simple-chronos-socket.h"

using namespace boost::python;


BOOST_PYTHON_MODULE(chronos)
{
  // we are going to mess with GIL, the infamous Global Interpreter Lock for python.
  // so we have to add this statement 
  PyEval_InitThreads();

  class_<SimpleChronosSocket>("SimpleChronosSocket", init<std::string, object>())
    .def("publish_string", &SimpleChronosSocket::publishString)
    .def("remove", &SimpleChronosSocket::remove)
    .def("callback", &SimpleChronosSocket::passCallback)
  ;
}
